opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -92,24 +93,21 @@ namespace Opm {
             ADB              mob;   // Phase mobility (per cell)
         };
 
-        struct SimulatorData {
+        struct SimulatorData : public Opm::FIPDataEnums {
             SimulatorData(int num_phases);
 
-            enum FipId {
-                FIP_AQUA = Opm::Water,
-                FIP_LIQUID = Opm::Oil,
-                FIP_VAPOUR = Opm::Gas,
-                FIP_DISSOLVED_GAS = 3,
-                FIP_VAPORIZED_OIL = 4,
-                FIP_PV = 5,                    //< Pore volume
-                FIP_WEIGHTED_PRESSURE = 6
-            };
+            using Opm::FIPDataEnums :: FipId ;
+            using Opm::FIPDataEnums :: fipValues ;
+
             std::vector<ReservoirResidualQuant> rq;
             ADB rsSat; // Saturated gas-oil ratio
             ADB rvSat; // Saturated oil-gas ratio
-            std::array<V, 7> fip;
+
+            std::array<V, fipValues> fip;
         };
 
+        typedef Opm::FIPData FIPDataType;
```

Comment:
What is the typedef needed for? If it was not used, much of the changed code could remain using FIPData with no change I think.